docs/adding-a-new-app.md

```diff
@@ -1,248 +1,178 @@
-# Adding a new app to GOV.UK
+[rds-database-management]: https://github.com/alphagov/govuk-aws/blob/master/doc/guides/rds-database-management.md
+[govuk-secrets]: https://github.com/alphagov/govuk-secrets
 
-This Puppet repository defines all the apps which run on GOV.UK.
+# Adding a new app to GOV.UK
 
 Create a new file in `modules/govuk/manifests/apps` named `my_app.pp`:
 
 ```
-# Be sure to include documentation of the class at the top of the file, as
-# per https://docs.puppetlabs.com/guides/style_guide.html#puppet-doc.
+# == Class: govuk::apps::myapp
+#
+# Read more: https://github.com/alphagov/myapp
+#
+# === Parameters
+# [*port*]
+#   What port should the app run on? Find the next free one in development-vm/Procfile
+#
+# [*enabled*]
+#   Whether to install the app. Don't change the default (false) until production-ready
+#
+# [*secret_key_base*]
+#   The key for Rails to use when signing/encrypting sessions (in govuk-secrets)
+#
+# [*sentry_dsn*]
+#   The app-specific URL used by Sentry to report exceptions (in govuk-secrets)
+#
+# [*oauth_id*]
+#   The OAuth ID used to identify the app to GOV.UK Signon (in govuk-secrets)
+#
+# [*oauth_secret*]
+#   The OAuth secret used to authenticate the app to GOV.UK Signon (in govuk-secrets)
+#
+# [*db_hostname*]
+#   The hostname of the database server to use for in DATABASE_URL environment variable
+#
+# [*db_username*]
+#   The username to use for the DATABASE_URL environment variable
+#
+# [*db_password*]
+#   The password to use for the DATABASE_URL environment variable
 #
-# You may wish to copy an existing app, eg. calculators:
-# https://github.com/alphagov/govuk-puppet/blob/3e926f0d/modules/govuk/manifests/apps/calculators.pp
+# [*db_name*]
+#   The database name to use for the DATABASE_URL environment variable
 #
-class govuk::apps::my_app (
-  $port = 3456,
+class govuk::apps::myapp (
+  $port = 99999999,
+  $enabled = false,
+  $secret_key_base = undef,
+  $sentry_dsn = undef,
+  $oauth_id = undef,
+  $oauth_secret = undef,
+  $db_hostname = undef,
+  $db_password = undef,
+  $db_name = 'myapp_production',
 ) {
-  govuk::app { 'my-app':
+  $app_name = 'myapp'
+
+  $ensure = $enabled ? {
+    true  => 'present',
+    false => 'absent',
+  }
+
+  # see modules/govuk/manifests/app.pp for more options
+  govuk::app { $app_name:
+    ensure            => $ensure,
     app_type          => 'rack',
     port              => $port,
     sentry_dsn        => $sentry_dsn,
     vhost_ssl_only    => true,
-    health_check_path => '/',
+    health_check_path => '/healthcheck', # must return HTTP 200 for an unauthenticated request
+    deny_framing      => true,
+    asset_pipeline    => true,
   }
-}
-```
 
-The `health_check_path` must be an endpoint on the application returning a
-HTTP 200 status for an unauthenticated request. Monitoring checks for the application are
-configured using this path and fail if the request isn't successful.
+  govuk::app::envvar {
```

Comment:
If I remember right doesn't this need to have capital letter prefixes to set the defaults

e.g: https://github.com/alphagov/govuk-puppet/blob/master/modules/govuk/manifests/apps/email_alert_api.pp#L155-L159

based on this wonderfully intuitive syntax: https://puppet.com/docs/puppet/5.3/lang_defaults.html

Reply:
Ah, I assumed there wasn't a difference :-|